core/src/main/scala/cats/instances/function.scala

```diff
@@ -76,13 +76,13 @@ private[instances] sealed trait Function1Instances extends Function1Instances0 {
         }
     }
 
-
-  implicit val catsStdInstancesForFunction1: Choice[Function1] with CommutativeArrow[Function1] =
-    new Choice[Function1] with CommutativeArrow[Function1] {
-      def choice[A, B, C](f: A => C, g: B => C): Either[A, B] => C = {
-        case Left(a)  => f(a)
-        case Right(b) => g(b)
-      }
+  implicit val catsStdInstancesForFunction1: ArrowChoice[Function1] with CommutativeArrow[Function1] =
+    new ArrowChoice[Function1] with CommutativeArrow[Function1] {
+      def choose[A, B, C, D](f: A => C)(g: B => D): Either[A, B] => Either[C, D] =
+        _.fold(
```

Comment:
using patten match rather than `fold` has performance gain. see https://github.com/typelevel/cats/issues/1951

Reply:
Yup, can do. I tend to forget that end of things.